src/lf_apply.rs

```diff
@@ -52,70 +53,60 @@ unsafe fn backup_lpf<BD: BitDepth>(
-                slice::from_raw_parts_mut(
-                    &mut *dst.offset(BD::pxstride(dst_stride as usize * 2) as isize)
-                        as *mut BD::Pixel,
-                    dst_w as usize,
-                ),
-                slice::from_raw_parts(
-                    &mut *dst
-                        .offset(BD::pxstride(dst_stride as usize * (top + 2) as usize) as isize)
-                        as *mut BD::Pixel,
-                    dst_w as usize,
-                ),
-                dst_w as usize,
-            );
-            BD::pixel_copy(
-                slice::from_raw_parts_mut(
-                    &mut *dst.offset(BD::pxstride(dst_stride as usize * 3) as isize)
-                        as *mut BD::Pixel,
-                    dst_w as usize,
-                ),
-                slice::from_raw_parts(
-                    &mut *dst
-                        .offset(BD::pxstride(dst_stride as usize * (top + 3) as usize) as isize)
-                        as *mut BD::Pixel,
-                    dst_w as usize,
-                ),
-                dst_w as usize,
-            );
+            if dst_stride < 0 {
+                let mut dst_offset_delta1 = 3 * BD::pxstride(-dst_stride as usize);
+                let dst_tmp = dst.split_at_mut(dst_offset - dst_offset_delta1);
+                let mut dst_offset_delta0 =
+                    dst_offset - top as usize * BD::pxstride(-dst_stride as usize);
+                dst_tmp.1[dst_offset_delta1..][..dst_w as usize]
+                    .copy_from_slice(&dst_tmp.0[dst_offset_delta0..][..dst_w as usize]);
+                dst_offset_delta0 -= BD::pxstride(-dst_stride as usize);
+                dst_offset_delta1 -= BD::pxstride(-dst_stride as usize);
+                dst_tmp.1[dst_offset_delta1..][..dst_w as usize]
+                    .copy_from_slice(&dst_tmp.0[dst_offset_delta0..][..dst_w as usize]);
+                dst_offset_delta0 -= BD::pxstride(-dst_stride as usize);
+                dst_offset_delta1 -= BD::pxstride(-dst_stride as usize);
+                dst_tmp.1[dst_offset_delta1..][..dst_w as usize]
+                    .copy_from_slice(&dst_tmp.0[dst_offset_delta0..][..dst_w as usize]);
+                dst_offset_delta0 -= BD::pxstride(-dst_stride as usize);
+                dst_offset_delta1 -= BD::pxstride(-dst_stride as usize);
+                dst_tmp.1[dst_offset_delta1..][..dst_w as usize]
+                    .copy_from_slice(&dst_tmp.0[dst_offset_delta0..][..dst_w as usize]);
+            } else {
+                let mut dst_offset_delta = 0;
+                let dst_tmp = dst.split_at_mut(
+                    (dst_offset as isize
+                        + top as isize * BD::pxstride(dst_stride as usize) as isize)
+                        as usize,
+                );
+                dst_tmp.0[dst_offset + dst_offset_delta..][..dst_w as usize]
+                    .copy_from_slice(&dst_tmp.1[dst_offset_delta..][..dst_w as usize]);
```

Comment:
For these `.copy_from_slice`s, can we still use `BD::pixel_copy`?  `fn pixel_copy` does `dst[..n].copy_from_slice(&src[..n])`, which is what you're doing with `dst_w` anyways.